README.md

```diff
@@ -227,6 +227,45 @@ jobs:
           workdir: example/
 ```
 
+## Artifacts
+
+Each call to this action produces one metadata artifact as well as either one or two tarballs
+containing a tagged single-arch docker image.
+
+### Metadata: Tag List
+
+The metadata file is a JSON blob containing three lists of tags generated by the call to the action.
+The file is named either `docker_tag_list_${TARGET}.json` or `docker_tag_list_${TARGET}_redhat.json`.
+
+If you're defining `redhat_tag` then the filename will have the `_redhat` suffix.
+
+This tag list is used by downstream processes to quickly determine which tags were generated by
+the build, so they know which remote registries they'll require credentials for when uploading.
+
+Note that when calling this action in a matrix, each call will produce the same tag list file,
+because its name and contents are not dependent on the os and/or architecture being built.
```

Comment:
Ah, the answer is that these are the user-facing tags, which are not arch dependent because the docker client gets a manifest list using that tag and then decides on which actual image to pull. I'll add a note in the text to explain.